src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/ErrorFixerAgent.cs

```diff
@@ -0,0 +1,142 @@
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _adminClient = adminClient ?? throw new ArgumentNullException(nameof(adminClient));
+            _agent = new AsyncLazy<PersistentAgent>(CreateAgentAsync);
+        }
+
+        public async Task FixCodeAsync(CancellationToken ct)
+        {
+            PersistentAgent agent = await _agent.GetValueAsync(ct).ConfigureAwait(false);
+
+            // TODO: Implement the code fixing logic here
+            await Task.CompletedTask.ConfigureAwait(false);
+        }
+
+        private async Task<PersistentAgent> CreateAgentAsync(CancellationToken ct)
+        {
+            _logger.LogInformation("Creating AZC Fixer agent...");
+            Response<PersistentAgent> response = await _adminClient.CreateAgentAsync(
+                model: _appSettings.Model,
+                name: _appSettings.AgentName,
+                instructions: _appSettings.AgentInstructions,
+                tools: new[] { new FileSearchToolDefinition() },
+                cancellationToken: ct).ConfigureAwait(false);
+
+            PersistentAgent agent = response.Value;
+            if (agent == null || string.IsNullOrEmpty(agent.Id))
+            {
+                throw new InvalidOperationException("Failed to create AZC Fixer agent");
+            }
+
+            _logger.LogInformation("✅ Agent created successfully: {Name} ({Id})", agent.Name, agent.Id);
+            return agent;
+        }
+
+        private async Task DeleteAgentsAsync(CancellationToken ct)
+        {
+            List<PersistentAgent> agents = new List<PersistentAgent>();
+
+            await foreach (PersistentAgent agent in _adminClient.GetAgentsAsync(cancellationToken: ct))
+            {
+                agents.Add(agent);
+            }
+
+            IEnumerable<Task> deleteTasks = agents.Select(async agent =>
+            {
+                try
+                {
+                    _logger.LogInformation("Deleting agent: {Name} ({Id})", agent.Name, agent.Id);
+                    await _adminClient.DeleteAgentAsync(agent.Id, ct).ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Failed to delete agent {Name} ({Id})", agent.Name, agent.Id);
+                }
+            });
+
+            await Task.WhenAll(deleteTasks).ConfigureAwait(false);
+        }
+
+        public async ValueTask DisposeAsync()
+        {
+            if (_agent.IsValueCreated)
+            {
+                try
+                {
+                    await DeleteAgentsAsync(CancellationToken.None).ConfigureAwait(false);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Error deleting agents during disposal");
+                }
+            }
+        }
+    }
+
+    internal class AsyncLazy<T>
```

Comment:
Let's remove this and use `Lazy<T>` and the synchronous [CreateAgent](https://learn.microsoft.com/dotnet/api/azure.ai.agents.persistent.persistentagentsadministrationclient.createagent?view=azure-dotnet) version.    The async path isn't going to save us anything and adds a ton of complexity.